sir_cake/store/utils.py

```diff
@@ -0,0 +1,70 @@
+from django.contrib import messages
+from django.core.exceptions import ObjectDoesNotExist
+
+from seller.models import Item, Order
+from users.models import AnonymousUser
+from .models import BasketItem
+
+
+def add_one_item_to_basket(request):
+
+    if anonymous_user_without_session(request):
+        return 'choose_purchasing_mode'
+
+    try:
+        item_to_buy = Item.objects.get(id=request.POST.get('item_pk'))
+    except ObjectDoesNotExist:
+        messages.error(request, "Item does not exist")
+        return 'store'
+
+    # Check how many items are 'reserved'
+    num_reserved_items = 0
+    for basket_instance in BasketItem.objects.filter(item_to_buy=item_to_buy):
+        num_reserved_items += basket_instance.quantity
+
+    if item_to_buy.quantity - num_reserved_items <= 0:
+        messages.error(request, "Item out of stock")
```

Comment:
Maybe a better error message would be `Not enough items in stock given the selected quantity`